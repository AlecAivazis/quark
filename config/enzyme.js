// external imports
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import browserEnv from 'browser-env'

// add a request animation frame polyfill
import raf from 'raf'
raf.polyfill()

// enable the react 16 adapter for enzyme
Enzyme.configure({ adapter: new Adapter() })

// add various browser mocks
browserEnv()
