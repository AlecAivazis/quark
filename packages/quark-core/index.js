export * from './components';
import * as _styles from './styles';
export { _styles as styles };