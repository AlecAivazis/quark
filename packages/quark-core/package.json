{
    "name": "quark-core",
    "version": "1.0.6",
    "module": "index.js",
    "main": "index.js",
    "dependencies": {
        "react-native-web": "0.3.4",
        "svgs": "^3.1.1"
    },
    "peerDependencies": {
        "react": "^16.3.0"
    }
}
