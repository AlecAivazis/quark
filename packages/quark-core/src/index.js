// @flow
export * from './components'
export * as styles from './styles'
