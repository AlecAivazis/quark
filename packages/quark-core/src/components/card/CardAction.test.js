// external imports
import React from 'react'
import { Text } from 'react-native'
import { mount } from 'enzyme'
// local imports
import { CardAction } from 'quark-core'

import renderer from 'react-test-renderer'

it('renders', () => {
    expect(
        mount(
            <CardAction>
                <Text>hello</Text>
            </CardAction>
        )
    ).toMatchSnapshot()
})
