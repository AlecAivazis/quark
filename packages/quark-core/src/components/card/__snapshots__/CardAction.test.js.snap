// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<CardAction>
  <GetTheme>
    <ButtonWithText
      activeColor="#F9F9F9"
      constrainSize={false}
      defaultColor="white"
      size="medium"
      style={
        Object {
          "alignSelf": "flex-end",
          "borderRadius": 0,
          "borderStyle": "solid",
          "borderTopColor": "#EAEAEE",
          "borderTopWidth": 1,
          "flex": 1,
        }
      }
      textColor="#4A90E2"
    >
      <BaseButton
        activeColor="#F9F9F9"
        constrainSize={false}
        defaultColor="white"
        size="medium"
        style={
          Object {
            "alignSelf": "flex-end",
            "borderRadius": 0,
            "borderStyle": "solid",
            "borderTopColor": "#EAEAEE",
            "borderTopWidth": 1,
            "flex": 1,
          }
        }
      >
        <TouchableWithoutFeedback
          accessible={true}
          onPressIn={[Function]}
          onPressOut={[Function]}
        >
          <AnimatedComponent
            accessible={true}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-end",
                "backgroundColor": "rgba(255, 255, 255, 1)",
                "borderRadius": 0,
                "borderStyle": "solid",
                "borderTopColor": "#EAEAEE",
                "borderTopWidth": 1,
                "cursor": "pointer",
                "flex": 1,
                "height": 40,
                "justifyContent": "center",
                "overflow": "hidden",
                "touchAction": "manipulation",
              }
            }
          >
            <View
              accessible={true}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-end",
                  "backgroundColor": "rgba(255, 255, 255, 1)",
                  "borderRadius": 0,
                  "borderStyle": "solid",
                  "borderTopColor": "#EAEAEE",
                  "borderTopWidth": 1,
                  "cursor": "pointer",
                  "flex": 1,
                  "height": 40,
                  "justifyContent": "center",
                  "overflow": "hidden",
                  "touchAction": "manipulation",
                }
              }
            >
              <div
                className="rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "WebkitAlignItems": "center",
                    "WebkitAlignSelf": "flex-end",
                    "WebkitBoxAlign": "center",
                    "WebkitBoxPack": "center",
                    "WebkitFlex": 1,
                    "WebkitFlexGrow": "1 !important",
                    "WebkitFlexShrink": "1 !important",
                    "WebkitJustifyContent": "center",
                    "alignItems": "center",
                    "alignSelf": "flex-end",
                    "backgroundColor": "rgba(255, 255, 255, 1)",
                    "borderBottomLeftRadius": "0px",
                    "borderBottomRightRadius": "0px",
                    "borderTopColor": "rgba(234,234,238,1)",
                    "borderTopLeftRadius": "0px",
                    "borderTopRightRadius": "0px",
                    "borderTopWidth": "1px",
                    "cursor": "pointer",
                    "flex": 1,
                    "flexGrow": "1 !important",
                    "flexShrink": "1 !important",
                    "height": "40px",
                    "justifyContent": "center",
                    "msFlex": 1,
                    "msFlexAlign": "center",
                    "msFlexItemAlign": "end",
                    "msFlexNegative": "1 !important",
                    "msFlexPack": "center",
                    "msFlexPositive": "1 !important",
                    "msTouchAction": "manipulation",
                    "overflowX": "hidden",
                    "overflowY": "hidden",
                    "touchAction": "manipulation",
                  }
                }
                tabIndex="0"
              >
                <Text
                  style={
                    Object {
                      "color": "#4A90E2",
                      "fontSize": 16,
                    }
                  }
                >
                  <InnerText
                    style={
                      Object {
                        "color": "#4A90E2",
                        "fontSize": 16,
                      }
                    }
                  >
                    <GetTheme>
                      <Text
                        style={
                          Object {
                            "color": "#4A90E2",
                            "fontSize": 16,
                            "margin": 0,
                          }
                        }
                      >
                        <div
                          className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-ubezar rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
                          dir="auto"
                          style={
                            Object {
                              "color": "rgba(74,144,226,1)",
                            }
                          }
                        >
                          <Text>
                            <span
                              className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-poiln3 rn-fontSize-7cikom rn-fontVariant-ebii48 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-irrty rn-wordWrap-qvutc0"
                              dir="auto"
                            >
                              hello
                            </span>
                          </Text>
                        </div>
                      </Text>
                    </GetTheme>
                  </InnerText>
                </Text>
              </div>
            </View>
          </AnimatedComponent>
        </TouchableWithoutFeedback>
      </BaseButton>
    </ButtonWithText>
  </GetTheme>
</CardAction>
`;
